v0/destinations/kustomer/transform.js

```diff
@@ -0,0 +1,194 @@
+const get = require("get-value");
+const axios = require("axios");
+const { EventType } = require("../../../constants");
+const logger = require("../../../logger");
+const {
+  CONFIG_CATEGORIES,
+  MAPPING_CONFIG,
+  BASE_ENDPOINT
+} = require("./config");
+const {
+  defaultRequestConfig,
+  getFieldValueFromMessage,
+  constructPayload,
+  removeUndefinedAndNullValues,
+  defaultPostRequestConfig,
+  defaultPutRequestConfig
+} = require("../../util");
+const { validateEvent } = require("./util");
+
+// Function responsible for constructing the Kustomer (User) Payload for identify
+// type of events.
+const constructKustomerPayload = (message, category) => {
+  const kustomerPayload = constructPayload(
+    message,
+    MAPPING_CONFIG[category.name]
+  );
+
+  if (
+    !get(kustomerPayload, "name") &&
+    getFieldValueFromMessage(message, "firstName") &&
+    getFieldValueFromMessage(message, "lastName")
+  ) {
+    kustomerPayload.name = `${getFieldValueFromMessage(
+      message,
+      "firstName"
+    )} ${getFieldValueFromMessage(message, "lastName")}`;
+  }
+
+  if (getFieldValueFromMessage(message, "email")) {
+    kustomerPayload.emails = kustomerPayload.emails
+      ? kustomerPayload.emails
+      : [{ type: "home", email: getFieldValueFromMessage(message, "email") }];
+  }
+
+  if (getFieldValueFromMessage(message, "phone")) {
+    kustomerPayload.phones = kustomerPayload.phones
+      ? kustomerPayload.phones
+      : [{ type: "home", phone: getFieldValueFromMessage(message, "phone") }];
+  }
+  const url = get(message, "traits.website")
+    ? get(message, "traits.website")
+    : get(message, "context.traits.website");
+  if (url) {
+    kustomerPayload.urls = [
+      {
+        url
+      }
+    ];
+  }
+
+  const address = getFieldValueFromMessage(message, "address");
+  if (address) {
+    kustomerPayload.locations = [
+      {
+        type: "home",
+        address:
+          typeof address === "string"
+            ? address
+            : `${address.street}, ${address.city}, ${address.state}, ${address.postalCode}`
```

Comment:
Can we handle it in a different way? Otherwise, it might be a field filled with `undefined`

Reply:
fixed